### PR TITLE
Added Lenses to resolve #46

### LINKED ISSUE #46: Add missing lenses
Missing Lenses:
Sigma 18-300mm F3.5-6.3 DC MACRO OS HSM | Contemporary 014
Canon EF-S24mm f/2.8 STM

I don't know enough rust to find an answer, but what is the purpose of the toml table full of lenses? Why not just copy the information from existing exif data? Are the other parameters used somewhere I haven't seen in the code?

### PR DESCRIPTION
**Pull request by pfjason**
- Canon EF-S24mm f/2.8 STM
	- Sigma 18-300mm F3.5-6.3 DC MACRO OS HSM | Contemporary 014

### CODE REVIEW

**Comment by pfjason** [comment]
Resolves #46

File: rawler/data/lenses.toml [hunk 1]
```diff
@@ -158,6 +158,13 @@ aperture_range = [[14, 10], [14, 10]]
 
 # Canon EF-S mount lenses
 
+[[lenses]]
+key = "EF-S24mm f/2.8 STM"
+make = "Canon"
+model = "EF-S24mm f/2.8 STM"
+focal_range = [[24, 1], [24, 1]]
+aperture_range = [[28, 10], [22, 1]]
```
Comment (cytrinox) [inline rawler/data/lenses.toml:166]: This should be 28, 10, 28, 10 as only the widest aperture is relevant.

File: rawler/data/lenses.toml [hunk 2]
```diff
@@ -158,6 +158,13 @@ aperture_range = [[14, 10], [14, 10]]
 
 # Canon EF-S mount lenses
 
+[[lenses]]
+key = "EF-S24mm f/2.8 STM"
+make = "Canon"
+model = "EF-S24mm f/2.8 STM"
```
Comment (cytrinox) [inline rawler/data/lenses.toml:164]: The correct model name is: "EF-S 24mm f/2.8 STM"

File: rawler/data/lenses.toml [hunk 3]
```diff
@@ -204,6 +211,13 @@ model = "150-600mm f/5-6.3 DG OS HSM | C"
 focal_range = [[150, 1], [600, 1]]
 aperture_range = [[50, 10], [63, 10]]
 
+[[lenses]]
+key = "18-300mm F3.5-6.3 DC MACRO OS HSM | Contemporary 014"
+make = "Sigma"
+model = "18-300mm F3.5-6.3 DC MACRO OS HSM | Contemporary 014"
```
Comment (cytrinox) [inline rawler/data/lenses.toml:217]: Correct model is "18-300mm f/3.5-6.3 DC Macro OS HSM | C"

**Review by cytrinox — CHANGES_REQUESTED** [review]
Thanks for adding these lenses. I've added the correct model names, could you please change and rebase your PR?

File: rawler/data/lenses.toml (hunk 1 above)
Comment (pfjason) [reply rawler/data/lenses.toml]: fixed

File: rawler/data/lenses.toml (hunk 2 above)
Comment (pfjason) [reply rawler/data/lenses.toml]: fixed

File: rawler/data/lenses.toml (hunk 3 above)
Comment (pfjason) [reply rawler/data/lenses.toml]: fixed

Comment (cytrinox) [reply rawler/data/lenses.toml]: Still wrong, it's f/3.5 instead of F3.5 :wink:

Comment (pfjason) [reply rawler/data/lenses.toml]: woops... fixed

**Pull request merged**